pb/paste/views.py

```diff
@@ -37,7 +38,82 @@ def _url(endpoint, **kwargs):
 
 @paste.route('/')
 def index():
-    return render_template("index.html")
+    http_user_header = current_app.config.get('CREATE_PASTE_USER_HEADER')
+    alias_whoami = current_app.config.get('ALIAS_WHOAMI')
+    return render_template("index.html", http_user_header=http_user_header, alias_whoami=alias_whoami)
+
+@paste.route('/whoami')
+def whoami():
+    user = None
+    header_name = current_app.config.get('REMOTE_USER_HEADER')
+    if header_name:
+        user = request.headers.get(header_name)
+    return jsonify({"user": user})
+
+def _parse_query_cursor():
+    defaultValue = None
+    try:
+        value = int(request.args["cursor"])
+        return datetime.fromtimestamp(value / 1000)
+    except Exception:
+        return defaultValue
+
+def _search_pastes():
+    query = {}
+    for key in request.args:
+        if key in ("cursor", "date"):
+            continue
+        # Multiple values for a query param mean logical OR.
+        values = request.args.getlist(key)
+        if len(values) == 1:
+            query[key] = values[0]
+        else:
+            query[key] = {"$in": values}
+
+    cursor = _parse_query_cursor()
+    if cursor:
+        query["date"] = {"$lt": cursor}
+
+    results = model.get_meta_without_content(**query)
+    results = results.sort("date", DESCENDING)
+
+    # Query 1 extra paste so we can tell if there are more pages.
+    limit = 100
+    results = results.limit(limit + 1)
+
+    pastes = [{**paste, "url": any_url(paste)} for paste in results]
+
+    nextCursor = None
+    if pastes and len(pastes) > limit:
+        pastes.pop()
+        nextCursor = int(pastes[-1]["date"].timestamp() * 1000)
+
+    return pastes, nextCursor
+
+@paste.route('/search')
+def search():
+    pastes, nextCursor = _search_pastes()
+
+    nextUrl = None
+    if nextCursor:
+        params = request.args.copy()
+        params["cursor"] = str(nextCursor)
+        nextUrl = f"{request.path}?{urlencode(params)}"
+
+    if request.accept_mimetypes.best == "application/json":
+        return jsonify({
+            "pastes": pastes,
+            "_links": {
+                "self": request.full_path,
+                "next": nextUrl,
+            },
+        })
+
+    response = make_response(render_template("search.html", results=pastes, nextUrl=nextUrl))
```

Comment:
This should not be the default; only reply with html when the Accept header negotiation results in text/html